apps/api/src/services/course/clone.ts

```diff
@@ -299,63 +139,93 @@ export async function cloneCourse(
   newDescription?: string,
   newSlug?: string,
   organizationId?: string
-): Promise<Course> {
-  const { data: course, error } = await fetchCourseData(courseId);
+): Promise<TCourse> {
+  // 1. fetch old course
+  const [course] = await getCourseById(courseId);
+
+  // 2. create group
+  const [newGroup] = await createGroup({
+    name: newTitle,
+    description: newDescription ?? course.description,
+    organizationId: organizationId
+  });
 
-  if (error) {
-    console.error('Error when cloning', error);
-    throw error;
-  }
+  // 3. create course for that group
+  // Explicitly select only the fields we want to copy (exclude id, createdAt, updatedAt)
+  const [newCourse] = await createCourse({
+    title: newTitle,
+    description: course.description,
+    overview: course.overview,
+    groupId: newGroup.id,
+    isTemplate: course.isTemplate,
+    logo: course.logo,
+    slug: newSlug ?? null,
+    metadata: course.metadata,
+    cost: course.cost,
+    currency: course.currency,
+    bannerImage: course.bannerImage,
+    isPublished: course.isPublished,
+    isCertificateDownloadable: course.isCertificateDownloadable,
+    certificateTheme: course.certificateTheme,
+    status: course.status,
+    type: course.type,
+    version: course.version
+  });
 
-  if (!course) {
-    throw new Error('Course not found');
-  }
+  // 4. add group member
+  await addGroupMember({
+    profileId: userId,
+    email: '',
+    groupId: newGroup.id,
+    roleId: 2
```

Comment:
we now have constants for `ROLE`.

`import { ROLE } from '@cio/utils/constants';`